Implementing the "are you sure?" plugin to display a confirmation mes…

nglasl opened the pull request:
…sage when navigating away from a partially completed form.

tractorcow commented:
Is there a way we can localise this message? 

When you bind the plugin, can you use the i18n JS library to set the custom messages? I only ask because we use userforms on some sites which require localisation.

tractorcow commented on javascript/UserForm.js:
```diff
@@ -721,6 +721,9 @@ jQuery(function ($) {
 		}, 180 * 1000);
+
+		// Bind a confirmation message when navigating away from a partially completed form.
+		$('form.userform').areYouSure();
```
```js
$('form.userform').areYouSure({
	message: ss.i18n._t('UserForms.LEAVE_CONFIRMATION', 'You have unsaved changes')
});
```

nglasl replied:
Thanks for the quick response! This is indeed possible.

I will make that change soon and let you know.

tractorcow replied:
I'll get you to add the master string to the en.js file too, please, if that's ok. :) there's a javascript/lang/src/en.js and a javascript/lang/en.js file to update. If you update just the master (en.js) that's sufficient.

nglasl commented:
This has now been updated to support localisation, @tractorcow.

tractorcow commented:
Sorry to be a bug, but I was thinking more about this last night.

If someone doesn't like this feature, and wants to turn it off, how can they do that? :P Is there a flag / config they can set? It probably wouldn't need to be CMS editable, just a config static would probably be enough.

nglasl commented:
I've now added a configuration flag and some further validation to ensure this doesn't cause any problems. I have also added a little something to the README, and rebased against upstream @tractorcow :)

nglasl commented:
I just thought I would check in to see how this is looking?

tractorcow commented:
It looks merged to me.

nglasl commented:
Awesome, thanks!
